### DISCUSSION TITLE
Issue with the current [@types/express] upgrade

### DISCUSSION THREAD

**Discussion by mbvgua** [DefinitelyTyped/DefinitelyTyped#71444] [Issues with a @types package] [answered]
Hey guys. I installed the most recent @types/express package that is currently at 5.0.0 and was working with it alongside the most recent express package that is currently at 4.21.2... The was this overload bug that kept bugging me ever since I created my first route. 

The app worked anyway so I did not give it much thought. But the more routes I add, the more verbose the errors became upto a point where I could no longer communicate with my endpoints. This is a snippet of how the message were: 
```bash
[build]   The last overload gave the following error.
[build]     Argument of type '(request: Request<ParamsDictionary, any, any, ParsedQs, Record<string, any>>, response: Response<any, Record<string, any>>) => Promise<...>' is not assignable to parameter of type 'Application<Record<string, any>>'.
[build]       Type '(request: Request<ParamsDictionary, any, any, ParsedQs, Record<string, any>>, response: Response<any, Record<string, any>>) => Promise<...>' is missing the following properties from type 'Application<Record<string, any>>': init, defaultConfiguration, engine, set, and 63 more.
[build] src/api-v1/routes/auth.routes.ts(7,28): error TS2769: No overload matches this call.
[build]   The last overload gave the following error.
[build]     Argument of type '(request: Request<{ id: string; }, any, any, ParsedQs, Record<string, any>>, response: Response<any, Record<string, any>>) => Promise<...>' is not assignable to parameter of type 'Application<Record<string, any>>'.
[build]       Type '(request: Request<{ id: string; }, any, any, ParsedQs, Record<string, any>>, response: Response<any, Record<string, any>>) => Promise<...>' is missing the following properties from type 'Application<Record<string, any>>': init, defaultConfiguration, engine, set, and 63 more.
```

Eventually, after several hours of debugging, on a whim, I decided to downgrade my current version of @types/express, and on doing so the error immediately vanished. I think that there is a new bug in the latest version that does not allow compatibility with the current release of express. It would be great if this was fixed.

[a comment by an automated account is not shown]

**Comment by jakebailey** [comment, marked as answer]
> Hey guys. I installed the most recent @types/express package that is currently at 5.0.0 and was working with it alongside the most recent express package that is currently at 4.21.2... 

If you're using `express@4`, use `@types/express@4`. If you're using `express@5`, use `@types/express@5`. I'm not sure why you would want to mix and match.

**Comment by mbvgua** [comment]
Oh my bad. I had copy pasted my node modules directory from an old project I was working on, hence it was still at `@express@4`, then installed `@types/express@5` afresh in the new directory and this caused the errors. 

Thanks for heads up!
